### PR TITLE
Modifications for IT8613E (BIOSTAR Group TB250-BTC) 

### PR DESCRIPTION
**Pull request by andreychernyshev**
modified:   it87.c for IT8613E support

### CODE REVIEW

File: it87.c [hunk 1]
```diff
@@ -557,6 +559,17 @@ static const struct it87_devices it87_devices[] = {
 		.num_temp_limit = 3,
 		.peci_mask = 0x07,
 	},
+	[it8613] = {
+                .name = "it8613",
+                .suffix = "E",
+                .features = FEAT_NEWER_AUTOPWM | FEAT_12MV_ADC | FEAT_16BIT_FANS
+                  | FEAT_TEMP_OFFSET | FEAT_TEMP_PECI | FEAT_SIX_FANS
```
Comment (groeck) [inline it87.c]: Per my information, the chip supports four fans and pwm controls. The first fan tachometer at 0x18/0x0d is not supported, and there is no tachometer at 0x4c/0x4d nor at 0x94/0x93.
ADC resolution, as mentioned earlier, is 11 mV.

File: it87.c [hunk 2]
```diff
@@ -557,6 +559,17 @@ static const struct it87_devices it87_devices[] = {
...
+                  | FEAT_IN7_INTERNAL | FEAT_SIX_PWM | FEAT_PWM_FREQ2
+                  | FEAT_SIX_TEMP | FEAT_VIN3_5V | FEAT_SCALING
```
Comment (groeck) [inline it87.c]: I have no indication that FEAT_VIN3_5V would apply here.

File: it87.c [hunk 3]
```diff
@@ -557,6 +559,17 @@ static const struct it87_devices it87_devices[] = {
...
+                  | FEAT_FANCTL_ONOFF,
```
Comment (groeck) [inline it87.c]: To my knowledge, FEAT_FANCTL_ONOFF is not supported.

File: it87.c [hunk 4]
```diff
@@ -2952,7 +2968,7 @@ static int __init it87_find(int sioaddr, unsigned short *address,
 
 		sio_data->beep_pin = superio_inb(sioaddr,
 						 IT87_SIO_BEEP_PIN_REG) & 0x3f;
-	} else if (sio_data->type == it8620 || sio_data->type == it8628 ||
+	} else if (sio_data->type == it8613 || sio_data->type == it8620 || sio_data->type == it8628 ||
```
Comment (groeck) [inline it87.c]: it8622 might be a better fit. However, as mentioned before, pwm1 and fan1 need to be disabled.

File: it87.c (hunk 1 above)
Comment (andreychernyshev) [inline it87.c]: BIOSTAR Group TB250-BTC have a support just only for two fan
fan2 for CPU fan 
fan3 for CASE fan

File: it87.c (hunk 3 above)
Comment (andreychernyshev) [inline it87.c]: need to double check

File: it87.c (hunk 4 above)
Comment (andreychernyshev) [inline it87.c]: ok 
this good

File: it87.c (hunk 1 above)
Comment (andreychernyshev) [inline it87.c]: fan4 and fan5 not used on this MB

Comment (andreychernyshev) [inline it87.c]: for info ,,,
it8613-isa-0a30
Adapter: ISA adapter
in0:          +0.06 V  (min =  +3.04 V, max =  +3.01 V)
in1:          +1.30 V  (min =  +2.27 V, max =  +2.96 V)
in2:          +2.16 V  (min =  +0.91 V, max =  +2.29 V)
+5V:          +4.32 V  (min =  +0.00 V, max =  +5.42 V)
in4:          +2.23 V  (min =  +0.95 V, max =  +2.68 V)
in5:          +3.06 V  (min =  +2.46 V, max =  +1.14 V)
in6:          +1.18 V  (min =  +3.02 V, max =  +1.14 V)
3VSB:         +6.02 V  (min =  +2.71 V, max =  +6.07 V)
Vbat:         +3.43 V  
fan2:        1083 RPM  (min = 1000 RPM)
fan3:        1591 RPM  (min = 1000 RPM)
fan4:           0 RPM  (min =   -1 RPM)
fan5:           0 RPM  (min =   -1 RPM)
temp1:        +41.0°C  (low  =  +0.0°C, high = +98.0°C)  sensor = thermistor
temp2:        +37.0°C  (low  =  -1.0°C, high = +127.0°C)  sensor = thermistor
temp3:        +39.0°C  (low  = +127.0°C, high = +127.0°C)
temp4:        +41.0°C  
temp5:        +41.0°C  
temp6:        +41.0°C  
intrusion0:  ALARM

Comment (andreychernyshev) [inline it87.c]: direct RPM managment forking fine !
echo 150 > pwm2  changing fan speed

File: it87.c (hunk 3 above)
Comment (andreychernyshev) [inline it87.c]: any way , i'll change

File: it87.c [hunk 5]
```diff
@@ -557,6 +559,16 @@ static const struct it87_devices it87_devices[] = {
 		.num_temp_limit = 3,
 		.peci_mask = 0x07,
 	},
+        [it8613] = {
+                .name = "it8613",
+                .suffix = "E",
+                .features = FEAT_NEWER_AUTOPWM | FEAT_12MV_ADC | FEAT_16BIT_FANS
```
Comment (groeck) [inline it87.c]: I was told that the ADC resolution is 11 mV, and have no reason to believe otherwise.

File: it87.c [hunk 6]
```diff
@@ -557,6 +559,16 @@ static const struct it87_devices it87_devices[] = {
...
+                  | FEAT_TEMP_OFFSET | FEAT_TEMP_PECI | FEAT_FIVE_FANS
```
Comment (groeck) [inline it87.c]: Wonder if the chip even supports five fans, or only three. I'll need to check with my sources ( I don't have the datasheet myself, and BIOSTAR is not responsive).

File: it87.c (hunk 1 above)
Comment (groeck) [inline it87.c]: 3VSB and +5V are obviously wrong. Vbat is too high, but that is due to the 11 vs. 12 mV ADC.

Comment (groeck) [inline it87.c]: The lack of a temp type for temp3 suggests that the chip may be multi-page.

File: it87.c (hunk 5 above)
Comment (andreychernyshev) [inline it87.c]: ok

Comment (andreychernyshev) [inline it87.c]: done

File: it87.c (hunk 6 above)
Comment (andreychernyshev) [inline it87.c]: unfortunately i not have a chip spec . current motherboard support 3 fan's sockets, but sysfan1 and sysfan2 - running in parallel on PWM signal.  RPM signal taked only from sysfan1 socket  
sensors utility - recognise CPU fan as fan2 ; sysfan1/sysfan2 as fan3
thx

File: it87.c (hunk 3 above)
Comment (andreychernyshev) [inline it87.c]: done

File: it87.c (hunk 4 above)
Comment (andreychernyshev) [inline it87.c]: done

File: it87.c (hunk 5 above)
Comment (andreychernyshev) [inline it87.c]: please look sensors response after changes (3.3V looks good ):
it8613-isa-0a30
Adapter: ISA adapter
in0:          +0.76 V  (min =  +2.78 V, max =  +2.76 V)
in1:          +1.19 V  (min =  +2.08 V, max =  +2.72 V)
in2:          +1.98 V  (min =  +0.84 V, max =  +2.10 V)
in3:          +1.98 V  (min =  +0.00 V, max =  +2.49 V)
in4:          +2.05 V  (min =  +0.87 V, max =  +2.45 V)
in5:          +2.81 V  (min =  +2.25 V, max =  +1.04 V)
in6:          +1.08 V  (min =  +2.77 V, max =  +1.04 V)
3VSB:         +5.52 V  (min =  +2.49 V, max =  +5.57 V)
Vbat:         +3.15 V  
+3.3V:        +3.30 V  
fan2:        1081 RPM  (min = 1000 RPM)
fan3:        1979 RPM  (min = 1000 RPM)
fan4:           0 RPM  (min =   -1 RPM)
fan5:           0 RPM  (min =   -1 RPM)
temp1:        +42.0°C  (low  =  +0.0°C, high = +98.0°C)  sensor = thermistor
temp2:        +34.0°C  (low  =  -1.0°C, high = +127.0°C)  sensor = thermistor
temp3:        +41.0°C  (low  = +127.0°C, high = +127.0°C)
intrusion0:  ALARM

File: it87.c (hunk 1 above)
Comment (andreychernyshev) [inline it87.c]: temp3: +39.0°C (low = +127.0°C, high = +127.0°C) 

yes. maybe multi-page .. but without spec we cant know.

Comment (groeck) [inline it87.c]: Why not ? You could either test it manually by selecting page 2 and checking if there are registers there, or you could select multi-page, assume that the chip is compatible to another in that regard (say, 8655 or 8665), and see if it reports reasonable temperature sensor types.

File: README [hunk 7]
```diff
@@ -6,6 +6,10 @@ Supported chips:
     Prefix: 'it8603'
     Addresses scanned: from Super I/O config space (8 I/O ports)
     Datasheet: Not publicly available
+  * IT8713E
+    Prefix: 'it8713'
```
Comment (jmaggard10) [inline README]: it8713 or it8613?

**Comment by groeck** [comment]
Pulled, squashed into one commit, and fixed up with information received.

**Pull request closed without merging**